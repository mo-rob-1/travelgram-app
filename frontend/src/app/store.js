import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import imageReducer from "../features/images/imageSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    images: imageReducer,
  },
});
