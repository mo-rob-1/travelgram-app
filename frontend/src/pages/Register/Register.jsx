import { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { register, reset } from "../../features/auth/authSlice";
import { useNavigate } from "react-router-dom";
import { toast } from "react-toastify";

function Register() {
  const [image, setImage] = useState(null);
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");

  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleChange = (e) => {
    setImage(e.target.files[0]);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (password !== confirmPassword) {
      toast.error("Passwords do not match");
      return;
    } else {
      const formData = new FormData();
      formData.append("image", image);
      formData.append("email", email);
      formData.append("password", password);
      formData.append("name", name);
      formData.append("confirmPassword", confirmPassword);

      dispatch(register(formData));

      navigate("/");
    }
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <input type="file" onChange={handleChange} required />
          {/* <button type="submit">Upload</button> */}
        </div>
        <div>
          <input type="text" placeholder="name" required value={name} onChange={(e) => setName(e.target.value)} />
        </div>
        <div>
          <input type="text" placeholder="email" required value={email} onChange={(e) => setEmail(e.target.value)} />
        </div>
        <div>
          <input
            type="password"
            placeholder="password"
            value={password}
            required
            onChange={(e) => setPassword(e.target.value)}
          />
        </div>
        <div>
          <input
            type="password"
            placeholder="confirm password"
            value={confirmPassword}
            required
            onChange={(e) => setConfirmPassword(e.target.value)}
          />
        </div>
        <button type="submit">Register</button>
      </form>
    </div>
  );
}

export default Register;
