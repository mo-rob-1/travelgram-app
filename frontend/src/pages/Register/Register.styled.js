import styled from "styled-components";
import { Link } from "react-router-dom";

export const Container = styled.div`
  min-height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Input = styled.input`
  width: 100%;
  margin-top: 4px;
  padding: 0.6rem;
  border: 1px solid #ccc;
`;

export const Button = styled.button`
  width: 100%;
  background-color: #222;
  color: #fff;
  border: none;
  padding: 0.8rem 0.8rem;
  cursor: pointer;
`;

export const HeadingWrapper = styled.div`
  text-align: center;
  margin-bottom: 2.8rem;
`;

export const Heading = styled.h1`
  margin-top: 0;
`;

export const FieldWrapper = styled.div`
  margin-bottom: 1rem;
`;

export const FormWrapper = styled.div`
  box-shadow: rgba(0, 0, 0, 0.1) 0px 1px 3px 0px, rgba(0, 0, 0, 0.06) 0px 1px 2px 0px;
  background-color: #fff;
  padding: 5rem 3rem;
  border-radius: 0.6rem;
`;

export const LinkWrapper = styled.div`
  text-align: center;
  margin-top: 2rem;
`;

export const StyledLink = styled(Link)`
  color: #222;
`;

export const ChooseFile = styled.input`
  width: 50%;
  margin-bottom: 1rem;
  @media (min-width: 768px) {
    width: 100%;
  }
`;
