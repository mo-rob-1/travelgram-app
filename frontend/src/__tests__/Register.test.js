import { render, screen } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import Register from "../pages/Register/Register";

describe("Register", () => {
  it("renders the register user page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders every Register text", () => {
    render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );
    expect(screen.getAllByText("Register")).toBeTruthy();
  });

  it("renders email input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );

    const emailInput = getByTestId("email-input");
    expect(emailInput).toBeTruthy();
  });

  it("renders password input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );

    const passwordInput = getByTestId("password-input");
    expect(passwordInput).toBeTruthy();
  });

  it("renders confirm password input", () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );

    const passwordInput = getByTestId("confirm-password-input");
    expect(passwordInput).toBeTruthy();
  });

  it("renders register button", () => {
    const { getByRole } = render(
      <Provider store={store}>
        <Router>
          <Register />
        </Router>
      </Provider>
    );

    const { getComputedStyle } = window;
    window.getComputedStyle = (elt) => getComputedStyle(elt);
    getByRole("button", { name: /register/i });
  });
});
