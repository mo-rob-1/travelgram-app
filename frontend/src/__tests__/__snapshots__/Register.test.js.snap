// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders the register user page 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Register
    </h1>
    <form>
      <div>
        <input
          required=""
          type="file"
        />
      </div>
      <div>
        <input
          data-testid="name-input"
          placeholder="name"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          data-testid="username-input"
          placeholder="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          data-testid="email-input"
          placeholder="email"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          data-testid="password-input"
          placeholder="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div>
        <input
          data-testid="confirm-password-input"
          placeholder="confirm password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        data-testid="register-button"
        type="submit"
      >
        Register
      </button>
    </form>
  </div>
</DocumentFragment>
`;
